### PR TITLE
Fully process clang opts in execution phase

### PR DESCRIPTION
**Pull request by brentleyjones**
This works around a Bazel OOM from too large of a params file:

```
FATAL: bazel ran out of memory and crashed. Printing stack trace:
java.lang.OutOfMemoryError
	at java.base/java.io.ByteArrayOutputStream.hugeCapacity(Unknown Source)
	at java.base/java.io.ByteArrayOutputStream.grow(Unknown Source)
	at java.base/java.io.ByteArrayOutputStream.ensureCapacity(Unknown Source)
	at java.base/java.io.ByteArrayOutputStream.write(Unknown Source)
	at java.base/java.io.BufferedOutputStream.flushBuffer(Unknown Source)
	at java.base/java.io.BufferedOutputStream.write(Unknown Source)
	at com.google.devtools.build.lib.actions.ParameterFile.writeContentUtf8(ParameterFile.java:180)
	at com.google.devtools.build.lib.actions.ParameterFile.writeContent(ParameterFile.java:122)
	at com.google.devtools.build.lib.actions.ParameterFile.writeParameterFile(ParameterFile.java:112)
	at com.google.devtools.build.lib.actions.CommandLines$ParamFileActionInput.writeTo(CommandLines.java:319)
	at com.google.devtools.build.lib.remote.util.DigestUtil.compute(DigestUtil.java:71)
	at com.google.devtools.build.lib.remote.merkletree.DirectoryTreeBuilder.lambda$buildFromActionInputs$1(DirectoryTreeBuilder.java:153)
	at com.google.devtools.build.lib.remote.merkletree.DirectoryTreeBuilder.build(DirectoryTreeBuilder.java:264)
	at com.google.devtools.build.lib.remote.merkletree.DirectoryTreeBuilder.buildFromActionInputs(DirectoryTreeBuilder.java:147)
	at com.google.devtools.build.lib.remote.merkletree.DirectoryTreeBuilder.fromActionInputs(DirectoryTreeBuilder.java:82)
	at com.google.devtools.build.lib.remote.merkletree.DirectoryTreeBuilder.fromActionInputs(DirectoryTreeBuilder.java:68)
	at com.google.devtools.build.lib.remote.merkletree.MerkleTree.build(MerkleTree.java:230)
	at com.google.devtools.build.lib.remote.RemoteExecutionService.uncachedBuildMerkleTreeVisitor(RemoteExecutionService.java:457)
	at com.google.devtools.build.lib.remote.RemoteExecutionService.buildMerkleTreeVisitor(RemoteExecutionService.java:429)
	at com.google.devtools.build.lib.remote.RemoteExecutionService.lambda$buildInputMerkleTree$0(RemoteExecutionService.java:382)
	at com.google.devtools.build.lib.exec.SpawnInputExpander.walkNestedSetInputs(SpawnInputExpander.java:351)
	at com.google.devtools.build.lib.exec.SpawnInputExpander.walkInputs(SpawnInputExpander.java:310)
	at com.google.devtools.build.lib.remote.common.RemotePathResolver$DefaultRemotePathResolver.walkInputs(RemotePathResolver.java:116)
	at com.google.devtools.build.lib.remote.RemoteExecutionService.buildInputMerkleTree(RemoteExecutionService.java:377)
	at com.google.devtools.build.lib.remote.RemoteExecutionService.buildRemoteAction(RemoteExecutionService.java:520)
	at com.google.devtools.build.lib.remote.RemoteSpawnCache.lookup(RemoteSpawnCache.java:95)
	at com.google.devtools.build.lib.exec.AbstractSpawnStrategy.exec(AbstractSpawnStrategy.java:145)
	at com.google.devtools.build.lib.exec.AbstractSpawnStrategy.exec(AbstractSpawnStrategy.java:112)
	at com.google.devtools.build.lib.actions.SpawnStrategy.beginExecution(SpawnStrategy.java:47)
	at com.google.devtools.build.lib.exec.SpawnStrategyResolver.beginExecution(SpawnStrategyResolver.java:64)
	at com.google.devtools.build.lib.analysis.actions.SpawnAction.beginExecution(SpawnAction.java:352)
	at com.google.devtools.build.lib.actions.Action.execute(Action.java:133)
	at com.google.devtools.build.lib.skyframe.SkyframeActionExecutor$5.execute(SkyframeActionExecutor.java:957)
	at com.google.devtools.build.lib.skyframe.SkyframeActionExecutor$ActionRunner.continueAction(SkyframeActionExecutor.java:1124)
	at com.google.devtools.build.lib.skyframe.SkyframeActionExecutor$ActionRunner.run(SkyframeActionExecutor.java:1082)
	at com.google.devtools.build.lib.skyframe.ActionExecutionState.runStateMachine(ActionExecutionState.java:160)
	at com.google.devtools.build.lib.skyframe.ActionExecutionState.getResultOrDependOnFuture(ActionExecutionState.java:93)
	at com.google.devtools.build.lib.skyframe.SkyframeActionExecutor.executeAction(SkyframeActionExecutor.java:516)
	at com.google.devtools.build.lib.skyframe.ActionExecutionFunction.checkCacheAndExecuteIfNeeded(ActionExecutionFunction.java:827)
	at com.google.devtools.build.lib.skyframe.ActionExecutionFunction.computeInternal(ActionExecutionFunction.java:323)
	at com.google.devtools.build.lib.skyframe.ActionExecutionFunction.compute(ActionExecutionFunction.java:161)
	at com.google.devtools.build.skyframe.AbstractParallelEvaluator$Evaluate.run(AbstractParallelEvaluator.java:571)
	at com.google.devtools.build.lib.concurrent.AbstractQueueVisitor$WrappedRunnable.run(AbstractQueueVisitor.java:382)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
```

### CODE REVIEW

**Comment by brentleyjones** [comment]
I'll add some tests for this, to replace the tests in `process_compiler_opts_tests.bzl` I'm about to remove.

File: tools/params_processors/swift_debug_settings_processor.py [hunk 1]
```diff
@@ -14,13 +14,128 @@ def _build_setting_path(path):
...
+        return opt
+    elif opt.startswith("-fmodule-map-file="):
+        path = opt[18:]
+        if path == ".":
+            return "-fmodule-map-file=$(PROJECT_DIR)"
+        if _is_relative_path(path):
+            return "-fmodule-map-file=$(PROJECT_DIR)/" + path
+        return opt
+    elif opt.startswith("-iquote"):
+        path = opt[7:]
```
Comment (CognitiveDisson) [inline tools/params_processors/swift_debug_settings_processor.py:55]: Nit: `path = opt[len("-iquote"):]` it will be more readable.

Comment (brentleyjones) [reply tools/params_processors/swift_debug_settings_processor.py:55]: We know the length though, so I'm going to opt out of it for now. I can add a comment though.

File: tools/params_processors/swift_debug_settings_processor.py [hunk 2]
```diff
@@ -14,13 +14,128 @@ def _build_setting_path(path):
...
+        return None
+
+    if opt.startswith("-F"):
+        path = opt[2:]
+        if path == ".":
+            return "-F$(PROJECT_DIR)"
+        if _is_relative_path(path):
+            return "-F$(PROJECT_DIR)/" + path
+        return opt
+    elif opt.startswith("-fmodule-map-file="):
```
Comment (CognitiveDisson) [inline tools/params_processors/swift_debug_settings_processor.py:47]: As an idea:
```
path_clang_opts = [
"-F",
"-fmodule-map-file="
"-iquote"
]
for path_clang_opt in path_clang_opts:
    if opt.startswith(path_clang_opt):
        path = opt[len(path_clang_opt):]
        if not path:
            return opt
        if path == ".":
            return "-F$(PROJECT_DIR)"
        if _is_relative_path(path):
            return "-F$(PROJECT_DIR)/" + path
        return opt
```

**Pull request merged**